src/rules/no-named-as-default.js

```diff
@@ -15,28 +15,70 @@ module.exports = {
 
   create(context) {
     function checkDefault(nameKey, defaultSpecifier) {
+      /**
+       * For ImportDefaultSpecifier we're interested in the "local" name (`foo` for `import {bar as foo} ...`)
+       * For ExportDefaultSpecifier we're interested in the "exported" name (`foo` for `export {bar as foo} ...`)
+       */
+      const analyzedName = defaultSpecifier[nameKey].name;
+
       // #566: default is a valid specifier
-      if (defaultSpecifier[nameKey].name === 'default') { return; }
+      if (analyzedName === 'default') { return; }
 
       const declaration = importDeclaration(context, defaultSpecifier);
+      /** @type {import('../exportMap').default | null} */
+      const importedModule = ExportMapBuilder.get(declaration.source.value, context);
+      if (importedModule == null) { return; }
 
-      const imports = ExportMapBuilder.get(declaration.source.value, context);
-      if (imports == null) { return; }
+      if (importedModule.errors.length > 0) {
+        importedModule.reportErrors(context, declaration);
+        return;
+      }
 
-      if (imports.errors.length) {
-        imports.reportErrors(context, declaration);
+      if (!importedModule.hasDefault) {
+        // Impossible: how could we even get here? Kept in case of unexpected regressions.
```

Comment:
why would this be impossible? wouldn't a module that only has named exports qualify?

Reply:
The rule itself is triggered only for `ImportDefaultSpecifier`/`ExportDefaultSpecifier`. So the question is: is it possible for `{Import,Export}DefaultSpecifier` rule to be triggered if there are no default exports in the imported module?

I expect that... no? But I don't know. That would be improper code anyway, that's why we just skip the analysis with `return`

Reply:
it would definitely be code that fails at runtime, but it's entirely possible to write that code and have it parse :-)

Reply:
Added additional tests for this as well 3a0af7ecd2d2c1c1effbb6fcdbc3165c38e6c026

Reply:
Added additional tests for this as well in 7e2e78f8b33a19a47b660fba9e8ac2faf8822c6c